src/language-css/printer-postcss.js

```diff
@@ -493,77 +491,155 @@ function genericPrint(path, options, print) {
 
       return node.value;
     }
-    // postcss-values-parser
-    case "value-value":
-    case "value-root":
-      return print("group");
 
-    case "value-comment":
-      return options.originalText.slice(locStart(node), locEnd(node));
-
-    case "value-comma_group":
-      return printCommaSeparatedValueGroup(path, options, print);
+    // postcss-values-parser
+    // case "value":
+    // case "root":
+    //   return print("group");
+
+    // case "comment":
+    //   return options.originalText.slice(locStart(node), locEnd(node));
+
+    // case "comma_group":
+    //   return printCommaSeparatedValueGroup(path, options, print);
+
+    // case "paren_group":
+    //   return printParenthesizedValueGroup(path, options, print);
+
+    case "color":
+      return node.value.toString();
+
+    case "function-call":
+      return node.toString();
+
+    // case "func":
+    //   return [
+    //     node.value,
+    //     insideAtRuleNode(path, "supports") && isMediaAndSupportsKeywords(node)
+    //       ? " "
+    //       : "",
+    //     print("group"),
+    //   ];
+
+    case "list": {
+      // TODO: Every node needs a `type`
+      node.type = "list";
+      // TODO: Need a way to determine if this list is wrapped in parens
+      const hasParens = false;
+      const parentNode = path.parent;
+      const nodes = path.map(
+        ({ node }) => (typeof node === "string" ? node : print()),
+        "nodes",
+      );
+      // TODO: It looks like `url()` is just parsed as a `string`
+      if (
+        parentNode &&
+        isURLFunctionNode(parentNode) &&
+        (node.groups.length === 1 ||
+          (node.groups.length > 0 &&
+            node.groups[0].type === "value-comma_group" &&
+            node.groups[0].groups.length > 0 &&
+            node.groups[0].groups[0].type === "value-word" &&
+            node.groups[0].groups[0].value.startsWith("data:")))
+      ) {
+        return [hasParens ? "(" : "", join(",", nodes), hasParens ? ")" : ""];
+      }
+      if (!hasParens) {
+        const forceHardLine = path.match(
+          (node) =>
+            // TODO: The original `shouldBreakList()` checks for comma-groups
+            node.some((node) =>
+              ["list", "binary-operation"].includes(node.sassType),
+            ),
```

Comment:
postcss-values-parser breaks every value between two commas into its own "comma-group" if it's more than a single node, and uses that to determine line-breaks and indentation. With sass-parser each item in a list is parsed individually, so this is an initial attempt to track a list of any sass-parser node types that postcss-values-parser would have considered more than one node (and therefore a "comma-group")

Reply:
Up to you in the end, but it's also possible that it's worth just rethinking the underlying logic of this formatting in the first place. Might be worth looking into how e.g. JS formats code in lists.